aiida_testing/mock_code/__init__.py

```diff
@@ -14,3 +14,6 @@
     "testing_config",
     "mock_code_factory",
 )
+
+# ensure aiida's pytest plugin is loaded, which we rely on
+pytest_plugins = ['aiida.manage.tests.pytest_fixtures']
```

Comment:
@ltalirz now users don't have to use `pytest_plugins`

Reply:
Hm... so defining the `pytest_plugins` variable in any part of the code will be picked up by `pytest`?

I read [here](https://docs.pytest.org/en/latest/reference/reference.html#globalvar-pytest_plugins) that it should be defined in test modules or the conftest file.

I would personally vote to let the users explicitly specify this, for my taste the fixtures are already "magic enough"

Reply:
but you literally cannot use the mock code fixtures without it, its essentially a dependency of the plugin

Reply:
IMO enforcing that every user has to specify `pytest_plugins` is just adding extra complication

Reply:
Yeah, with the magic insertion of the fixtures via the `pyproject.toml` it feels like an unnecessary burden (obviously that magic insertion is equivalent).

But I fear a bit that users will no longer know which fixtures are available in their tests, and where to look for them in the code...

Also, about the `pyproject.toml`, I'm wondering: I may want to run the test suite of other packages in my aiida environment - say, `pgsu`.
Since the `aiida_profile` fixture is `autouse=True`, does this then mean that it will be used in the test suite of my other non-AiiDA package as well?

Reply:
> Since the `aiida_profile fixture` is `autouse=True`,

Ha well I never quite got why this needed to be the case in the first place 😅
I believe you made it autouse in: https://github.com/aiidateam/aiida-core/pull/3319, is there reason why it has to be?

Reply:
> I may want to run the test suite of other packages in my aiida environment

IMO thats a very esoteric use case though; I use tox and thus have a different virtual environment created for every different test suite I run, I feel thats a better use of environments

> But I fear a bit that users will no longer know which fixtures are available in their tests, and where to look for them in the code...

you just run `pytest --fixtures` (as I've already put in the documentation here)

Reply:
I don't mind either way, probably easier to have them available automatically. One drawback is that people not very familiar with the test systems will maybe need more time to find out where fixtures come from. But the difference here between looking into the `conftest.py` or `pyproject.toml` for the test system dependencies is minimal.

But, do you know whether there is something like `pytest.importorskip` to skip tests when a fixtures are not available?

Reply:
Thanks for the feedback. In that case, let's keep the current implementation for the moment.

> But, do you know whether there is something like `pytest.importorskip` to skip tests when a fixtures are not available?

I don't know how to skip a test based on the availability of a fixture